src/ui/components/SecondaryToolbox/ReactDevTools.js

```diff
@@ -42,11 +49,17 @@ function InitReactDevTools() {
   wall = {
     emit() {},
     listen(listener) {
-      wall._listener = listener;
+      wall._listener = msg => {
+        try {
+          listener(msg);
+        } catch (err) {
+          console.warn("Error in ReactDevTools frontend", err);
+        }
+      };
     },
     async send(event, payload) {
       if (event == "inspectElement") {
-        if (inspected.has(payload.id)) {
+        if (inspected.has(payload.id) && !payload.path) {
```

Comment:
can you explain what is happening with `payload.path`? 

It looks like we might be able to add a guard at the top

```js
// <explain why we're skipping payload.path>
if (pathload.path) {
  return;
}
```

Reply:
Simply skipping if `payload.path` is set would break deep inspection of components' props and state: Only the top levels are initially transmitted to the ReactDevTools frontend and when you go deeper, more data is retrieved using an `inspectElement` message with `payload.path` set to identify the object that you want to inspect. The `inspected` Set should only be used for the `inspectElement` messages without `payload.path`.